### ISSUE TITLE
[TODO] Replace `get_headlines_info()` with a `get_headlines_as_df()` function, and use pandas

### ISSUE THREAD

**Issue by jonjoncardoso** [jonjoncardoso/lse-ds105-week09-experiment#1]
## Context:

In the current version of [Notebook 01](https://github.com/jonjoncardoso/ds105-fictional-sniffle/blob/main/notebooks/Notebook%2001%20-%20Data%20Collection.ipynb), the function `get_headlines_info()` is nice and all; it returns a list that can be readily converted to a pandas data frame, but it would be nicer if we used **vectorised operations** from the start.

**Current usage:**

```python
pd.DataFrame(get_headlines_info(headlines_soup)).head()
```

yields a nice tidy table:

|    | headline id                               | parent tag   | parent tag text                                 |
|---:|:------------------------------------------|:-------------|:------------------------------------------------|
|  0 | Overview                                  | h2           | Overview[edit]                                  |
|  1 | History_and_relationships_to_other_fields | h2           | History and relationships to other fields[edit] |
|  2 | Artificial_intelligence                   | h3           | Artificial intelligence[edit]                   |
|  3 | Data_mining                               | h3           | Data mining[edit]                               |
|  4 | Optimization                              | h3           | Optimization[edit]                              |

## Tasks

- [x] Replace `def get_headlines_info(headlines_soup)` with a `def get_headlines_as_df(headlines_soup)`
- [x] Instead of list append, propose a more `pandas`-style solution

**Comment by jonjoncardoso** [comment]
Looking for help 🆘🆘🆘

(_this is how you show you are desperate_)

**Comment by OjasSB** [comment]
My implementation:

```python
def get_headlines_as_df(headlines_soup):
    return pd.DataFrame({'headline id': [headline.get('id') for headline in headlines_soup], \
                        'parent tag': [headline.parent.name for headline in headlines_soup], \
                        'parent tag text': [headline.parent.text for headline in headlines_soup]})
```

**Comment by tz1211** [comment]
```python 
def get_headlines_as_df(headlines_soup):
    """
    Parses the information from the headlines on a Wikipedia page.

    Parameters
    ----------
    headlines_soup : list
        A list of BeautifulSoup objects with the class 'mw-headline'

    Returns
    -------
    df
        A pandas df with the following columns:
        - headline id
        - parent tag (h2, h3, h4, h5, h6, etc.)
        - parent tag text
    """

    parents = map(lambda x: x.parent, headlines_soup)
    headline_id = list(map(lambda x: x.get("id"), headlines_soup)) 
    parent_tag = list(map(lambda x: x.parent.name, headlines_soup))
    parent_tag_text = list(map(lambda x: x.parent.text, headlines_soup)) 
    df = pd.DataFrame({"headline id": headline_id, "parent tag": parent_tag, "parentg tag text": parent_tag_text}) 

    return df
    
```

**Comment by tz1211** [comment]
```python 
def get_headlines_as_df(headlines_soup):
    """
    Parses the information from the headlines on a Wikipedia page.

    Parameters
    ----------
    headlines_soup : list
        A list of BeautifulSoup objects with the class 'mw-headline'

    Returns
    -------
    df
        A pandas df with the following columns:
        - headline id
        - parent tag (h2, h3, h4, h5, h6, etc.)
        - parent tag text
    """

    parents = map(lambda x: x.parent, headlines_soup)
    headline_id = map(lambda x: x.get("id"), headlines_soup)
    parent_tag = map(lambda x: x.parent.name, headlines_soup)
    parent_tag_text = map(lambda x: x.parent.text, headlines_soup) 
    df = pd.DataFrame({"headline id": headline_id, "parent tag": parent_tag, "parentg tag text": parent_tag_text}) 

    return df
    
```

**Comment by jonjoncardoso** [comment]
Issue fixed in commit f6a1c35cdaed83b75f52d20b5ba12e3fd9fa4594
